File: backend/product-service/src/main/java/com/enterprise/ecommerce/product/service/ProductService.java
package com.enterprise.ecommerce.product.service;

import com.enterprise.ecommerce.product.dto.ProductRequest;
import com.enterprise.ecommerce.product.dto.ProductResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ProductService {

    Page<ProductResponse> getProducts(Pageable pageable, boolean includeInactive);

    ProductResponse getProductById(Long id, boolean includeInactive);

    Page<ProductResponse> searchProducts(String name, Pageable pageable, boolean includeInactive);

    List<ProductResponse> getProductsByCategory(Long categoryId, boolean includeInactive);

    ProductResponse createProduct(ProductRequest request);

    ProductResponse updateProduct(Long id, ProductRequest request);

    void deleteProduct(Long id);
}

File: frontend/src/services/productService.js
import axiosClient from '../api/axiosClient';

const unwrap = (response) => response.data.data;

export const productService = {
  getProducts: async (page = 0, size = 20) => {
    const response = await axiosClient.get('/api/products', {
      params: { page, size },
    });
    return unwrap(response);
  },

  getProductById: async (id) => {
    const response = await axiosClient.get(`/api/products/${id}`);
    return unwrap(response);
  },

  searchProducts: async (name, page = 0, size = 20) => {
    const response = await axiosClient.get('/api/products/search', {
      params: { name, page, size },
    });
    return unwrap(response);
  },

  getProductsByCategory: async (categoryId) => {
    const response = await axiosClient.get(`/api/products/category/${categoryId}`);
    return unwrap(response);
  },

  createProduct: async (productData) => {
    const response = await axiosClient.post('/api/products', productData);
    return unwrap(response);
  },

  updateProduct: async (id, productData) => {
    const response = await axiosClient.put(`/api/products/${id}`, productData);
    return unwrap(response);
  },

  deleteProduct: async (id) => {
    const response = await axiosClient.delete(`/api/products/${id}`);
    return response.data;
  },

  getCategories: async () => {
    const response = await axiosClient.get('/api/categories');
    return unwrap(response);
  },

  createCategory: async (categoryData) => {
    const response = await axiosClient.post('/api/categories', categoryData);
    return unwrap(response);
  },

  updateCategory: async (id, categoryData) => {
    const response = await axiosClient.put(`/api/categories/${id}`, categoryData);
    return unwrap(response);
  },

  deleteCategory: async (id) => {
    const response = await axiosClient.delete(`/api/categories/${id}`);
    return response.data;
  },
};
